resources/public/api/conf/2.0/application.yaml

```diff
@@ -2043,14 +2106,51 @@ components:
         self:
           $ref: '#/components/schemas/Self'
         departure:
-          $ref: '#/components/schemas/Departure'
+          $ref: '#/components/schemas/Departure2'
       example:
         self:
           href: /customs/transits/movements/departures/6365135ba5e821ee/messages
         departure:
           href: /customs/transits/movements/departures/6365135ba5e821ee
+    Links3:
+      title: Links3
+      properties:
+        self:
+          $ref: '#/components/schemas/Self1'
+        departure:
+          $ref: '#/components/schemas/Departure1'
+      example:
+        self:
+          href: /customs/transits/movements/departures/6365135ba5e821ee/messages
+        departure:
+          href: /customs/transits/movements/departures/6365135ba5e821ee
+    Links4:
+      title: Links4
+      required:
+        - self
+        - departure
+      type: object
+      properties:
+        self:
+          $ref: '#/components/schemas/Self'
+        departure:
+          $ref: '#/components/schemas/Departure1'
+      example:
+        self:
+          href: /customs/transits/movements/departures/6365135ba5e821ee/messages
+        departure:
+          href: /customs/transits/movements/departures/6365135ba5e821ee
+

```

Comment:
Links3 is for ``` Get all cached messages related to a departure ``` with making all the fields as optional as if passed movement does not contain any message then we need to return empty list. Links4 is for ``` Get a cached message related to a departure and message ID ``` with all the fields as mandatory. I looked into Self1 and Self2 and looks duplicating so I will remove one of them

Reply:
But `Links3` is still all mandatory within the context of the object it's in -- the links will still remain, it's just the departures/arrivals that sit alongside it, not within it.

Reply:
e.g. for below example messages are empty and that Links3 comes under that. Should it not be optional? so we would be having those details in this case?
``` {
"_links": {
"self": {
"href": "/customs/transits/movements/departures/6365135ba5e821ee/messages"
},
"departure": {
"href": "/customs/transits/movements/departures/6365135ba5e821ee"
}
},
"messages": []
}  ```

Reply:
No, `_links` is not optional, as `self` and `departure` hrefs are still valid.

Reply:
I think you are missing something here. See below example. Top Links is mandatory I agreed but Links inside the messages is going to be optional.

{
"_links": {
"self": {
"href": "/customs/transits/movements/departures/6365135ba5e821ee/messages"
},
"departure": {
"href": "/customs/transits/movements/departures/6365135ba5e821ee"
}
},
"messages": [
{
"_links": {
"self": {
"href": "/customs/transits/movements/departures/6365135ba5e821ee/message/634982098f02f00a"
},
"departure": {
"href": "/customs/transits/movements/departures/6365135ba5e821ee"
}
},
"id": "634982098f02f00a",
"departureId": "6365135ba5e821ee",
"received": "2022-11-10T15:32:51.459Z",
"type": "IE015",
"status": "Success"
}
]
}

Reply:
Ah, right. No, it's not optional there either because it's required on each departure in that list -- it's **required** there. You have a `departure` object, it will contain a `_links` object. Equally, the list of departures isn't optional either, it just might contain zero entries.

It's still not optional.